parmed/openmm/parameters.py

```diff
@@ -101,10 +107,20 @@ def from_parameterset(cls, params, copy=False):
         new_params.parametersets = params.parametersets
         new_params._combining_rule = params.combining_rule
         new_params.residues = params.residues
+        # Assign Lennard-Jones atom classes
+        ljtypes = dict()
+        counter = 0
+        for name, atom_type in iteritems(new_params.atom_types):
+            key = (atom_type.rmin, atom_type.epsilon, atom_type.rmin_14,
+                    atom_type.epsilon_14)
+            if key not in ljtypes:
+                ljtypes[key] = counter
```

Comment:
> gaff.dat defines separate LJ parameters for each atom type, even if they are duplicates. This means, as you observe, that using compressed numerical groups wouldn't let us mix-and-match due to collisions, but either the compressed types written as corresponding atom type names or writing them all by atom name would work.

True.  But now let's look at what else is allowed in Amber parameter files -- you can redefine L-J coefficients in an frcmod file for any atom type.  In our proposal, one of the atom types that shares L-J parameters with other types is defined as the "parent" of the L-J class.  If this atom type has its L-J parameters redefined in an frcmod file in some future force field (like the Garcia-Chen FF for nucleic acids), it could accidentally replace all of the L-J parameters for the whole group.

We're making this too complicated.  As I said, the only place where L-J type compression makes a difference is when you have NBFIXes and have to store lookup tables in shared memory.  But that is done via a `<script>` section where you can implement necessary compression in there.  And you would have to do this, anyway, since you don't want to include the parameters for atom types not in your system.

So I vote we either make class and type the same for these families of force fields, or just do away with class altogether (those two are identical, so I vote for the simplicity of the latter).